DUI3-DX/Connectors/ArcGIS/Speckle.Connectors.ArcGIS3/Bindings/ArcGISSendBinding.cs

```diff
@@ -57,6 +62,111 @@ private void SubscribeToArcGISEvents()
     StandaloneTablesRemovedEvent.Subscribe(GetIdsForStandaloneTablesRemovedEvent, true);
     MapPropertyChangedEvent.Subscribe(GetIdsForMapPropertyChangedEvent, true); // Map units, CRS etc.
     MapMemberPropertiesChangedEvent.Subscribe(GetIdsForMapMemberPropertiesChangedEvent, true); // e.g. Layer name
+
+    ActiveMapViewChangedEvent.Subscribe(SubscribeToMapMembersDataSourceChange, true);
+    LayersAddedEvent.Subscribe(GetIdsForLayersAddedEvent, true);
+    StandaloneTablesAddedEvent.Subscribe(GetIdsForStandaloneTablesAddedEvent, true);
+  }
+
+  private void SubscribeToMapMembersDataSourceChange(ActiveMapViewChangedEventArgs args)
+  {
+    var task = QueuedTask.Run(() =>
+    {
+      if (MapView.Active == null)
+      {
+        return;
+      }
+
+      // subscribe to layers
+      foreach (Layer layer in MapView.Active.Map.Layers)
+      {
+        if (layer is FeatureLayer featureLayer)
+        {
+          SubscribeToFeatureLayerDataSourceChange(featureLayer);
+        }
+      }
+      // subscribe to tables
+      foreach (StandaloneTable table in MapView.Active.Map.StandaloneTables)
+      {
+        SubscribeToTableDataSourceChange(table);
+      }
+    });
+    task.Wait();
+  }
+
+  private void SubscribeToFeatureLayerDataSourceChange(FeatureLayer layer)
+  {
+    if (SubscribedLayers.Contains(layer))
+    {
+      return;
+    }
+    Table layerTable = layer.GetTable();
+    SubscribeToAnyDataSourceChange(layerTable);
+    SubscribedLayers.Add(layer);
+  }
+
+  private void SubscribeToTableDataSourceChange(StandaloneTable table)
+  {
+    if (SubscribedTables.Contains(table))
+    {
+      return;
+    }
+    Table layerTable = table.GetTable();
+    SubscribeToAnyDataSourceChange(layerTable);
+    SubscribedTables.Add(table);
+  }
+
+  private void SubscribeToAnyDataSourceChange(Table layerTable)
+  {
+    RowCreatedEvent.Subscribe(
+      (args) =>
+      {
+        OnRowChanged(args);
+      },
+      layerTable
+    );
```

Comment:
Hmm, what if we have saved layer tables on the document? Should we subscribe each table at the initialization?

Reply:
Yes, it's called on the ActiveMapView change event, which means whenever we open a map all the saved layers will be subscribed

Reply:
Super!

Reply:
LayersAdded & TablesAdded events are catching up when we add layers to the document. Or did you mean something else?

Reply:
Ah yes, MapViewChanged subscribes to every layer existing in the Map which we have opened/ switched to. But definitely worth testing for edge cases

Reply:
Then we are good for now, when you are back, we can investigate edge cases.